### PR TITLE
fix(json): fix json dot path

### LINKED ISSUE #2175: improper handling of JSON path "."
**Describe the bug**
The current behavior is to treat a JSON path of "." as "$" for compatibility with tools like RedisInsight and official Redis libraries; however, this is different from the way Redis itself handles the legacy path ".".

**To Reproduce**

**Redis**

> 127.0.0.1:6379> JSON.SET doc $ '{"a":2}'
> OK
> 127.0.0.1:6379> JSON.GET doc
> {"a":2}
> 127.0.0.1:6379> JSON.GET doc $
> [{"a":2}]
> 127.0.0.1:6379> JSON.GET doc .
> {"a":2}

**Dragonfly**

> 127.0.0.1:6379> JSON.SET doc $ '{"a":2}'
> OK
> 127.0.0.1:6379> JSON.GET doc
> {"a":2}
> 127.0.0.1:6379> JSON.GET doc $
> [{"a":2}]
> 127.0.0.1:6379> JSON.GET doc .
> [{"a":2}]

**Expected behavior**
Providing a path of "." should give the same output as not providing a path at all. This would increase compatibility with RedisInsight and official Redis libraries.

I'm not as intimately familiar with other Redis libraries but redis-py automatically adds a path of "." if no path is provided. This is causing compatibility issues with using Dragonfly as a drop-in replacement since `redis.json().get("key")` is always returning a list due to the implied "." path.

### PR DESCRIPTION
**Pull request by dranikpg**
Fixes #2175, #2053

### CODE REVIEW

File: src/server/json_family.cc [hunk 1]
```diff
@@ -390,8 +391,9 @@ void SendJsonValue(ConnectionContext* cntx, const JsonType& j) {
 }
 
 OpResult<string> OpGet(const OpArgs& op_args, string_view key,
-                       vector<pair<string_view, JsonExpression>> expressions, bool should_format,
-                       const OptString& indent, const OptString& new_line, const OptString& space) {
+                       vector<pair<string_view, optional<JsonExpression>>> expressions,
```
Comment (romange) [inline src/server/json_family.cc:394]: I do not remember, expressions are passed by value by design?
if you are changing already, can you please rename to OpJsonGet?

Comment (dranikpg) [reply src/server/json_family.cc]: I just did touch that part, I refactored some of the bulky boilerplate parts. Changed, renamed (but all others don't have json 🤷🏻‍♂️)

**Review by romange — APPROVED** [review]

**Comment by romange** [comment]
Please double check with `redis.json().get("key")` that it works as expected as described in https://github.com/dragonflydb/dragonfly/issues/2175#issue-1993249353

**Comment by dranikpg** [comment]
I did, I added a testcase that . returns the exact document

**Pull request merged**